### PR TITLE
fix segfault

### PR DESCRIPTION
**Pull request by FernetMenta**
fixes segault introduced by https://github.com/xbmc/xbmc/pull/12213

### CODE REVIEW

**Comment by peak3d** [comment]
The message should only be called time by time (500ms is fine), so we should implement a timer action for this IMO.
Or, would be a post message ok if we leave it in ProcessSlow?

**Comment by FernetMenta** [comment]
- manipulation of gui items is supposed to be done in FrameMove
- if gui is not shown, a refresh message makes no sense
- timer for processSlow is not necessarly a good fit for updating gui items

all arguments why process slow is the wrong place

**Comment by peak3d** [comment]
pls. wait with merge, simply moving into FrameMove will not lead to the expected behaviour.
Have to check here

**Comment by FernetMenta** [comment]
read the code, there is a 500ms timer

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -2687,6 +2687,12 @@ void CApplication::FrameMove(bool processEvents, bool processGUI)
       m_skipGuiRender = true;
 #endif
 
+    if (g_advancedSettings.m_guiSmartRedraw && m_guiRefreshTimer.IsTimePast())
```
Comment (peak3d) [inline xbmc/Application.cpp]: Brackets about bit comparision missing

**Comment by peak3d** [comment]
I'm not sure if FrameMove is really called if GUI has nothing to do, thats the thing I wanted to test.
nevertheless thanx for the timer!

File: xbmc/Application.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/Application.cpp]: no brackets missing here, this is just a boolean and operation

**Comment by FernetMenta** [comment]
> I'm not sure if FrameMove is really called if GUI has nothing to do

I am

**Pull request merged**